apps/api/src/domain/models/media-item/commands/create-media-item.command.integration.spec.ts

```diff
@@ -0,0 +1,335 @@
+import { MIMEType } from '@coscrad/data-types';
+import setUpIntegrationTest from '../../../../app/controllers/__tests__/setUpIntegrationTest';
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { NotAvailable } from '../../../../lib/types/not-available';
+import { NotFound } from '../../../../lib/types/not-found';
+import { ArangoConnectionProvider } from '../../../../persistence/database/arango-connection.provider';
+import generateRandomTestDatabaseName from '../../../../persistence/repositories/__tests__/generateRandomTestDatabaseName';
+import TestRepositoryProvider from '../../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import { DTO } from '../../../../types/DTO';
+import InvalidEntityDTOError from '../../../domainModelValidators/errors/InvalidEntityDTOError';
+import MediaItemHasNoTitleInAnyLanguageError from '../../../domainModelValidators/errors/mediaItem/MediaItemHasNoTitleInAnyLanguageError';
+import getValidResourceInstanceForTest from '../../../domainModelValidators/__tests__/domainModelValidators/utilities/getValidResourceInstanceForTest';
+import { IIdManager } from '../../../interfaces/id-manager.interface';
+import { AggregateId } from '../../../types/AggregateId';
+import { ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import CommandExecutionError from '../../shared/common-command-errors/CommandExecutionError';
+import InvalidCommandPayloadTypeError from '../../shared/common-command-errors/InvalidCommandPayloadTypeError';
+import ResourceIdAlreadyInUseError from '../../shared/common-command-errors/ResourceIdAlreadyInUseError';
+import { assertCreateCommandError } from '../../__tests__/command-helpers/assert-create-command-error';
+import { assertCreateCommandSuccess } from '../../__tests__/command-helpers/assert-create-command-success';
+import { CommandAssertionDependencies } from '../../__tests__/command-helpers/types/CommandAssertionDependencies';
+import { MediaItem } from '../entities/media-item.entity';
+import { CreateMediaItem } from './create-media-item.command';
+import { CreateMediaItemCommandHandler } from './create-media-item.command-handler';
+
+const commandType = 'CREATE_MEDIA_ITEM';
+
+const buildValidCommandFSA = (id: AggregateId): FluxStandardAction<DTO<CreateMediaItem>> => ({
+    type: commandType,
+    payload: {
+        id,
+        title: 'Fishing Video',
+        titleEnglish: 'Fishing Video (Engl)',
+        contributions: [
+            {
+                contributorId: '101',
+                role: 'editor',
+            },
+            {
+                contributorId: '29',
+                role: 'executive producer',
+            },
+        ],
+        url: 'https://www.mysoundbox.org/vid.mp4',
+        mimeType: MIMEType.mp4,
+    },
+});
+
+const existingMediaItemId = `702096a0-c52f-488f-b5dc-22192e9aca3e`;
+
+const buildInvalidFSA = (
```

Comment:
We should hold off. It may be that we find a way to mock the `IdManager` so that the API difference between create and update commands goes away, in which case generalizing this will be much easier. For now, just hard-wire it in each test.